src/zenml/integrations/kubeflow/orchestrators/kubeflow_orchestrator.py

```diff
@@ -268,35 +297,52 @@ def up(self) -> None:
             return
 
         logger.info("Spinning up local Kubeflow Pipelines deployment...")
-        fileio.make_dirs(self.root_directory)
-        container_registry_port = int(container_registry.uri.split(":")[-1])
-        container_registry_name = self._get_k3d_registry_name(
-            port=container_registry_port
-        )
-        local_deployment_utils.write_local_registry_yaml(
-            yaml_path=self._k3d_registry_config_path,
-            registry_name=container_registry_name,
-            registry_uri=container_registry.uri,
-        )
-        local_deployment_utils.create_k3d_cluster(
-            cluster_name=self._k3d_cluster_name,
-            registry_name=container_registry_name,
-            registry_config_path=self._k3d_registry_config_path,
-        )
-        kubernetes_context = f"k3d-{self._k3d_cluster_name}"
-        local_deployment_utils.deploy_kubeflow_pipelines(
-            kubernetes_context=kubernetes_context
-        )
-        local_deployment_utils.start_kfp_ui_daemon(
-            pid_file_path=self._pid_file_path,
-            port=self.kubeflow_pipelines_ui_port,
-        )
 
-        logger.info(
-            f"Finished local Kubeflow Pipelines deployment. The UI should now "
-            f"be accessible at "
-            f"http://localhost:{self.kubeflow_pipelines_ui_port}/."
-        )
+        try:
+            fileio.make_dirs(self.root_directory)
+            container_registry_port = int(container_registry.uri.split(":")[-1])
+            container_registry_name = self._get_k3d_registry_name(
+                port=container_registry_port
+            )
+            local_deployment_utils.write_local_registry_yaml(
+                yaml_path=self._k3d_registry_config_path,
+                registry_name=container_registry_name,
+                registry_uri=container_registry.uri,
+            )
+            local_deployment_utils.create_k3d_cluster(
+                cluster_name=self._k3d_cluster_name,
+                registry_name=container_registry_name,
+                registry_config_path=self._k3d_registry_config_path,
+            )
+            kubernetes_context = f"k3d-{self._k3d_cluster_name}"
+            local_deployment_utils.deploy_kubeflow_pipelines(
+                kubernetes_context=kubernetes_context
+            )
+            local_deployment_utils.start_kfp_ui_daemon(
+                pid_file_path=self._pid_file_path,
+                port=self.kubeflow_pipelines_ui_port,
+            )
+            logger.info(
+                f"Finished local Kubeflow Pipelines deployment. The UI should now "
+                f"be accessible at "
+                f"http://localhost:{self.kubeflow_pipelines_ui_port}/."
+                f"The orchestrator is now up."
+            )
+        except Exception as e:
+            container_registry_port = int(container_registry.uri.split(":")[-1])
```

Comment:
Maybe we can move the duplicate code (getting the container registry, writing the yaml file) outside of the try block? I don't think that should fail anyway and if yes shouldn't be the reason that they have to spin it up manually

Reply:
Good idea! Thank you.